pylabrobot/resources/ml_star/mfx_modules.py

```
@@ -0,0 +1,228 @@
+""" ML Star MFX modules (including machine defintions placed on a MFX carrier) """
+
+from __future__ import annotations
+
+import logging
+from typing import List, Optional, Union
+
+from pylabrobot.resources.resource import Resource
+from pylabrobot.resources.carrier import Coordinate
+
+logger = logging.getLogger("pylabrobot")
+
+
+class MFXSite(Resource):
+  """ A single site within a MFX module. """
+
+  def __init__(self, name: str, size_x: float, size_y: float, size_z: float, spot: int,
+    category: str = "mfx_site", model: Optional[str] = None):
+    super().__init__(name=name, size_x=size_x, size_y=size_y, size_z=size_z, category=category,
+      model=model)
+    self.resource: Optional[Resource] = None
+    self.spot: int = spot
+
+  def assign_child_resource(
+    self,
+    resource: Resource,
+    location: Optional[Coordinate],
+    reassign: bool = True
+  ):
+    self.resource = resource
+    return super().assign_child_resource(resource, location=location)
```

I think we should raise an error in this case.